tests/cuda_ndarrays/test_cuda_ndarrays.cu

```diff
@@ -0,0 +1,505 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "ndarrays.h"
+#include "cuda_ndarrays.h"
+
+#define getname(X) #X
+
+void assert_double(double v1 , double v2, const char *dscr,
+        const char * func, const char *file, int32_t line)
+{
+    if (v1 != v2)
+    {
+        printf("[FAIL] %s:%d:%s\n", file, line, func);
+        printf("[INFO] %s:%f != %s:%f\n", getname(v1), v1, getname(v2),v2);
```

Comment:
Am I right that this is equivalent to:
```c
printf("[INFO] v1:%f != v2:%f\n", v1, v2);
```
?
Can you explain the advantage of using `getname` here?

Reply:
Yes, that's correct. Since the specific args v1 and v2 will always be named as such in the function, when the assertion fails, the printf statement using `getname` will always print "v1" and "v2". Updated here c899cf6873aa2ef11c2f9d53ce3874463a5f1415